controllers/pattern_controller.go

```diff
@@ -480,11 +468,63 @@ func (r *PatternReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		return err
 	}
 	r.driftWatcher, _ = newDriftWatcher(r.Client, mgr.GetLogger(), newGitClient())
+
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&api.Pattern{}).
 		Complete(r)
 }
 
+func (r *PatternReconciler) deployGitopsSubscription(filename string) error {
+
+	storedSubscription, _ := r.olmClient.getSubscription(gitopsSubscriptionName, subscriptionNamespace)
+	var newSpec *operatorv1alpha1.Subscription
+
+	// Check if the config map exists and read the config map values
+	if cm, err := r.fullClient.CoreV1().ConfigMaps(common.OperatorNamespace).Get(context.Background(), common.OperatorConfigFile, metav1.GetOptions{}); err == nil {
+		// Config Map exists
+		// Read config parameters
+
+		cmData := cm.Data
+
+		var installPlanApproval operatorv1alpha1.Approval
+
+		if cmData["gitops.installApprovalPlan"] == "Manual" {
+			installPlanApproval = operatorv1alpha1.ApprovalManual
+		} else {
+			installPlanApproval = operatorv1alpha1.ApprovalAutomatic
+		}
+
+		configSpec := operatorv1alpha1.SubscriptionSpec{
+			CatalogSource:          cmData["gitops.source"],
```

Comment:
I'd suggest:
```suggestion
			CatalogSource:          configValueWithDefault(cmData, "gitops.source", "whatever the default is")
```

Reply:
Implementation has been added to updates.